### PR TITLE
Add xml:Text() and annotations change

### PR DESCRIPTION
**Pull request by lasinicl**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: release-notes/swan-lake-alpha3-release-note.md [hunk 1]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 #### Language
 
+##### Improvements in annotations support
+
+`mapping-constructor-expr` is now allowed be optional in annotations, if the annotation type requires a map.
```
Comment (praneesha) [inline release-notes/swan-lake-alpha3-release-note.md:47]:
```suggestion
The `mapping-constructor-expr` is now allowed to be optional in annotations if the annotation type requires a map.
```

Comment (rdhananjaya) [reply release-notes/swan-lake-alpha3-release-note.md:47]: What we want to say is: annotation value is now allowed to be omitted from an annotation, if the annotation value is empty mapping constructor.
@MaryamZi  please help us word this.

Comment (praneesha) [reply release-notes/swan-lake-alpha3-release-note.md:47]: Let's simply say that. :)

"Now, it is optional to have the `mapping-constructor-expr` in an annotation if the annotation value is an empty mapping constructor."

Comment (MaryamZi) [reply release-notes/swan-lake-alpha3-release-note.md:47]: Shall we say something like

```suggestion
If the type of the annotation is a mapping type for which an empty mapping constructor is valid, the mapping constructor expression is no longer mandatory in the annotation attachment.

The absence of the mapping constructor expression in such an annotation attachment is equivalent to specifying a mapping constructor expression with no fields.
```

File: release-notes/swan-lake-alpha3-release-note.md [hunk 2]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+```ballerina
+type Annot record {|
+    int[] i = [];
+|};
+
+public annotation Annot v1 on function;
+
+@v1
```
Comment (MaryamZi) [inline release-notes/swan-lake-alpha3-release-note.md:55]:
```suggestion
@v1 // Same as `@v1 {}`
```

File: release-notes/swan-lake-alpha3-release-note.md [hunk 3]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
 
 #### Language
 
+##### Improvements in annotations support
```
Comment (MaryamZi) [inline release-notes/swan-lake-alpha3-release-note.md:45]:
```suggestion
##### Improvement to Annotation Attachment with Empty Mapping Constructor Expression
```

File: release-notes/swan-lake-alpha3-release-note.md [hunk 4]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+`mapping-constructor-expr` is now allowed be optional in annotations, if the annotation type requires a map.
+```ballerina
+type Annot record {|
+    int[] i = [];
+|};
+
+public annotation Annot v1 on function;
+
+@v1
+public function main(string... argv) {
```
Comment (MaryamZi) [inline release-notes/swan-lake-alpha3-release-note.md:56]:
```suggestion
public function main() {
```

File: release-notes/swan-lake-alpha3-release-note.md [hunk 5]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+}
+```
+
+##### New `xml:text()` function
```
Comment (MaryamZi) [inline release-notes/swan-lake-alpha3-release-note.md:60]: I think we'll have to add this under a separate langlib section.

```suggestion

##### Improved lang library functions

###### New `xml:text()` function
```

File: release-notes/swan-lake-alpha3-release-note.md [hunk 6]
```diff
@@ -42,6 +42,31 @@ If you have not installed Ballerina, then download the [installers](/downloads/#
...
+
+This method can be used to select all the items in a sequence that are of type `xml:Text`.
```
Comment (MaryamZi) [inline release-notes/swan-lake-alpha3-release-note.md:62]:
```suggestion
This function can be used to select all the items in a sequence that are of type `xml:Text`.
```

**Review by praneesha — APPROVED** [review]

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**